share/functions/__fish_is_zfs_feature_enabled.fish

```diff
@@ -0,0 +1,18 @@
+function __fish_is_zfs_feature_enabled -a feature target -d "Returns 0 if the given ZFS feature is available or enabled for the given full-path target (zpool or dataset), or any target if none given"
+	set -l pool (echo "$target" | cut -d '/' -f1)
+    set -l feature_name ""
+    set -l pattern "[[:space:]]"
+    if test -z "$pool"
+        set pattern "$pattern$feature$pattern"
+        set feature_name (zpool get all -H | grep "$pattern")
+    else
+        set pattern "$pool$pattern$feature$pattern"
+        set feature_name (zpool get all -H $pool | grep "$pattern")
```

Comment:
I'm unsure about that, because I had a hard time testing this piece of code; I remember it failed without this `else`, but why? That, I can't remember. The reason may have vanished in subsequent fish releases, though, as I use 2.4.0.

Reply:
You might have gotten the quoting wrong.

In the `set pattern`, you want `"$pool$pattern..."` inside quotes because you want $pool to expand to the empty string. For `zpool -H`, you want it _without_ quotes (i.e. `zpool get all -H $pool`, because you want it to expand to no argument.

Reply:
Oh, wait! Now I remember: without a target specified, I check any available pool, as, in this case, the function only checks if the feature is activated or enabled on the system. But, if a pool or a full-path target is specified, then the function does the check against this specific pool and, then, I have to find the matching line only for the specified pool. So, the `else` serves to narrow the `zfs` call to the specified pool if possible, thus decreasing processing time; I don't know if it's worth it, but it doesn't hurt.